pandas/parser.pyx

```diff
@@ -1151,8 +1181,15 @@ cdef class TextReader:
             # unicode variable width
             return self._string_convert(i, start, end, na_filter,
                                         na_hashset)
-
-
+        # is this comparison good enough?
+        elif dtype == '|O08':
```

Comment:
you don't at this point (caller does `.str`), but doesn't seem to be any good reason why, basically the original code - https://github.com/pydata/pandas/commit/3c77f2929c9a81fc8ceeab92d1b75fe41e273474